src/model/view.ts

```diff
@@ -0,0 +1,55 @@
+import { Node } from "./node"
+import { Evaluator } from "../core/eval"
+
+export type View = SegmentView
+
+export interface SegmentView {
+  type: "segment"
+  from: Node
+  to: Node
+  isVisible: boolean
+}
+
+export function segment(from: Node, to: Node, isVisible: boolean): SegmentView {
+  return {
+    type: "segment",
+    from,
+    to,
+    isVisible,
+  }
+}
+
+export function renderView(
+  view: View,
+  ev: Evaluator,
+  ctx: CanvasRenderingContext2D,
+  strokeColor: string,
+) {
+  switch (view.type) {
```

Comment:
For convenience, maybe we should do `ctx.save()` before the switch and `ctx.restore()` afterwards?